c10/xpu/XPUStream.h

```diff
@@ -0,0 +1,162 @@
+#pragma once
+
+#include <c10/core/Stream.h>
+#include <c10/xpu/XPUFunctions.h>
+
+namespace c10::xpu {
+
+/*
+ * Note [Stream Management]
+ *
+ * An XPUStream is an abstraction of an actual SYCL queue in which SYCL kernel
+ * can execute. Currently, there are several pools per device to manage SYCL
+ * queue, and a device's pool is lazily created.
+ *
+ * There are two pools per device. The first pool contains "normal priority"
+ * queues. The second pool is the "high priority" queues. There are 32 queues in
+ * per pool per device, and when a queue is requested one of these queues is
+ * returned round-robin. That is, the first queue requested is at index 0, the
+ * second at index 1... to index 31, then index 0 again.
+ *
+ * This means that if 33 queues are requested, the first and last queues
+ * requested are actually the same queue (under the covers) and kernels enqueued
+ * on them cannot run concurrently.
+ *
+ * It is safe to enqueue a kernel on the same queue from two different
+ * threads as the SYCL specification described.
+ */
+
+static constexpr int max_compile_time_stream_priorities = 2;
+
+/*
+ * This serves as a wrapper around c10::Stream and acts as a representation for
+ * a SYCL queue. On each device, a SYCL queue pool consists of kStreamsPerPool
+ * queues, and you can access a particular queue by its index. The index is
+ * extracted from XPUStream.id().
+ */
+class C10_XPU_API XPUStream {
+ public:
+  enum Unchecked { UNCHECKED };
```

Comment:
```suggestion
  enum class Unchecked { UNCHECKED };
```

Reply:
Done.

Reply:
If we add a class here, it seems a bit redundant to use `xpu::XPUStream::Unchecked::UNCHECKED` for invocation. So, removed keyword `class`.